plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergTableWithCustomLocation.java

```diff
@@ -56,10 +57,14 @@ protected DistributedQueryRunner createQueryRunner()
         metastoreDir = Files.createTempDirectory("test_iceberg").toFile();
         metastore = createTestingFileHiveMetastore(metastoreDir);
 
-        return IcebergQueryRunner.builder()
+        DistributedQueryRunner queryRunner = IcebergQueryRunner.builder()
                 .setIcebergProperties(Map.of("iceberg.unique-table-location", "true"))
                 .setMetastoreDirectory(metastoreDir)
                 .build();
+
+        fileSystem = getFileSystemFactory(queryRunner).create(SESSION);
```

Comment:
if this fails, will query runner get closed?

(using `@BeforeClass` avoids the question)